remotes/handlers.go

```diff
@@ -283,10 +283,6 @@ func SkipNonDistributableBlobs(f images.HandlerFunc) images.HandlerFunc {
 			return nil, images.ErrSkipDesc
 		}
 
-		if images.IsLayerType(desc.MediaType) {
```

Comment:
I noticed that the comments of the `SkipNonDistributableBlobs`
> // An example of this kind of content would be a Windows base layer, which is not supposed to be redistributed.

This pr looks good to me, but is there any problem in windows? @cpuguy83 PTAL, Thanks